src/core/lombok/eclipse/handlers/HandleEqualsAndHashCode.java

```diff
@@ -391,6 +441,20 @@ public MethodDeclaration createHashCode(EclipseNode type, Collection<Included<Ec
 			}
 		}
 		
+		/* this.$hashCodeCache = result; */ {
+			if (cacheHashCode) {
+				FieldReference hashCodeCacheRef = new FieldReference(HASH_CODE_CACHE_NAME_ARR, p);
+				hashCodeCacheRef.receiver = new ThisReference(pS, pE);
+				setGeneratedBy(hashCodeCacheRef, source);
+				setGeneratedBy(hashCodeCacheRef.receiver, source);
+				SingleNameReference resultRef = new SingleNameReference(RESULT, p);
+				setGeneratedBy(resultRef, source);
+				Assignment cacheResult = new Assignment(hashCodeCacheRef, resultRef, pE);
```

Comment:
```suggestion
				Assignment cacheResult = new Assignment(hashCodeCacheRef, resultRef, pE);
				cacheResult.sourceStart = pS; cacheResult.statementEnd = cacheResult.sourceEnd = pE;
```
Not really required because the `SetGeneratedByVisitor` sets these values too but thats what we do in all other cases.

Reply:
@Rawi01 I assume something similar should also be done for other places like return statements and other stuff I created for the eclipse path? Is there are clear criteria for determining for which types I should do this and for which I should not do this?

Reply:
I mentioned this case because the constructor only accepts the end position and `sourceStart` is not set. The return statement already accepts both positions.

In general, in most cases it is not required because of the visitor but the code still contains a lot of these source position things. Maybe on of the maintainers can clarify why we still set these values by hand.

Reply:
@Rawi01 got it. Thanks.